_posts/2017-08-10-snowplow-r91-stonehenge-released-emr-etl-runner-robustness.md

```diff
@@ -0,0 +1,192 @@
+---
+layout: post
+title-short: Snowplow 91 Stonehenge
+title: "Snowplow 91 Stonehenge released, EmrEtlRunner robustness"
+tags: [snowplow, emr]
+author: Ben
+category: Releases
+---
+
+We are pleased to issue the release of [Snowplow 91 Stonehenge][snowplow-release]. This release
+revolves around making EmrEtlRunner, the component launching the EMR steps for the batch pipeline,
+more robust. Most notably, this release fixes a long-standing bug in the way the staging step was
+performed.
+
+If you'd like to know more about R91 Stonehenge, named after
+[the prehistoric monument in England][stonehenge], read on!
+
+1. [Moving the staging step to S3DistCp](/blog/2017/08/10/snowplow-r91-stonehenge-released-emr-etl-runner-robustness#staging)
+2. [New EmrEtlRunner commands](/blog/2017/08/10/snowplow-r91-stonehenge-released-emr-etl-runner-robustness#commands)
+3. [New EmrEtlRunner CLI options](/blog/2017/08/10/snowplow-r91-stonehenge-released-emr-etl-runner-robustness#cli-options)
+4. [Upgrading](/blog/2017/08/10/snowplow-r91-stonehenge-released-emr-etl-runner-robustness#upgrading)
+5. [Roadmap](/blog/2017/08/10/snowplow-r91-stonehenge-released-emr-etl-runner-robustness#roadmap)
+6. [Help](/blog/2017/08/10/snowplow-r91-stonehenge-released-emr-etl-runner-robustness#help)
+
+![stonehenge][stonehenge-img]
+
+<!--more-->
+
+<h2 id="staging">1. Moving the staging step to S3DistCp</h2>
+
+This release overhauls the staging step which is in charge of moving data from the raw input buckets
+to a processing bucket to be further processed.
+
+This step used to run on the machine running EmrEtlRunner leveraging [Sluice][sluice]. Now it will
+run as an EMR step using [S3DistCp][s3-dist-cp].
+
+The main reason for the change is that the staging step used to rename files by transforming
+timestamps from a format to another. This renaming step produced duplicates in a multi-threaded
+environment which would result in files overwriting each other and would cause data loss.
+
+In our experiments, less than 1% of the total number of files were lost due to this issue.
+
+The fix introduced in this release delegates the staging step to S3DistCp and doesn't do any
+renaming.
```

Comment:
@alexanderdean I don't know how I should frame this, any help would be much appreciated.